javascript/javascript1/week4/homework/voiceAssistant.js

```diff
@@ -0,0 +1,155 @@
+      return `You've already told me your name is ${userName}`;
+    } else {
+      userName = name;
+      return `Nice to meet you ${userName}`;
+    }
+  }
+
+  // 2. "What is my name?"
+  if (cmdLower.startsWith("what is my name")) {
+    if (userName) {
+      return `Your name is ${userName}`;
+    } else {
+      return "I don't know your name yet.";
+    }
+  }
+
+  // 3. "Add fishing to my todo" (or similar)
+
+  if (cmdLower.startsWith("add ") && cmdLower.indexOf(" to my todo") !== -1) {
+    const startIndex = "add ".length;
+    const endIndex = cmdLower.indexOf(" to my todo");
+    const todoItem = cmd.substring(startIndex, endIndex).trim();
+    todos.push(todoItem);
+    return `${todoItem} added to your todo`;
+  }
+
+  // 4. "Remove fishing from my todo"
+  if (
+    cmdLower.startsWith("remove ") &&
+    cmdLower.indexOf(" from my todo") !== -1
+  ) {
+    const startIndex = "remove ".length;
+    const endIndex = cmdLower.indexOf(" from my todo");
+    const todoItem = cmd.substring(startIndex, endIndex).trim();
+
+    const index = todos.findIndex(
+      (item) => item.toLowerCase() === todoItem.toLowerCase()
+    );
+    if (index !== -1) {
+      todos.splice(index, 1);
+      return `Removed ${todoItem} from your todo`;
+    } else {
+      return `${todoItem} is not in your todo`;
+    }
+  }
+
+  // 5. "What is on my todo?"
+  if (cmdLower.indexOf("what is on my todo") !== -1) {
+    if (todos.length === 0) {
+      return "Your todo is empty.";
+    } else {
+      return `You have ${todos.length} todo${
+        todos.length > 1 ? "s" : ""
+      } - ${todos.join(" and ")}`;
+    }
+  }
+
+  // 6. "What day is it today?"
+  if (cmdLower.indexOf("what day is it today") !== -1) {
+    const now = new Date();
+    const day = now.getDate();
+    const monthNames = [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ];
+    const month = monthNames[now.getMonth()];
+    const year = now.getFullYear();
+    return `${day}. of ${month} ${year}`;
+  }
+
+  // 7. "What is 3 + 3?"
+  if (cmdLower.startsWith("what is ")) {
+    const expression = cmd.substring("what is ".length).trim();
+
+    const parts = expression.split(" ");
+    if (parts.length === 3) {
+      const num1 = parseFloat(parts[0]);
+      const operator = parts[1];
+      const num2 = parseFloat(parts[2]);
+      if (!isNaN(num1) && !isNaN(num2)) {
+        let result;
+        if (operator === "+") {
+          result = num1 + num2;
+        } else if (operator === "-") {
+          result = num1 - num2;
+        } else if (operator === "*") {
+          result = num1 * num2;
+        } else if (operator === "/") {
+          result = num1 / num2;
+        } else {
+          return "I can't do that math.";
+        }
+        return `${result}`;
```

Comment:
Nice work on this. Suggestion: 
- this is a nice case for using `switch`
- [return early pattern](https://forum.freecodecamp.org/t/the-return-early-pattern-explained-with-javascript-examples/19364)

```js
// If first condition isn't met, exit early
if (parts.length !== 3) {
	return ...
}

// Switch
    switch (operator) {
        case "+":
            result ...
            break;
        case "-":
            //...
        default:
            break;
```